src/setfit/integrations.py

```diff
@@ -1,28 +1,51 @@
 import importlib.util
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Optional
 
 from .utils import BestRun
+from .trainer import SetFitTrainer
 
 
-if TYPE_CHECKING:
-    from .trainer import SetFitTrainer
```

Comment:
Could we revert this? I prefer importing classes that are only required by typing under `if TYPE_CHECKING:`. This might also solve the test failures.

Reply:
@pneb I'm afraid the tests still fail, I think this is related.